test/lib/test-ringpop-cluster.js

```diff
@@ -127,12 +129,53 @@ function testRingpopCluster(opts, name, test) {
     }
 
     tape(name, function onTest(assert) {
-        var cluster = bootstrapClusterOf(opts, function onBootstrap(results) {
-            assert.on('end', function onEnd() {
-                destroyCluster(cluster);
+        var cluster = createClusterOf(opts);;
+        var joinResults;
+        var minProtocolPeriod = opts.minProtocolPeriod;
+        opts.minProtocolPeriod = 1;
+
+
+        function onConverged() {
+            // do not run onConverged if not all joins are succesful
+            if (joinResults === undefined) {
+                return;
+            }
+
+            if (!(opts.dontWaitForConvergence === true || opts.checkChecksums === false)) {
+                var checksums = _.uniq(_.pluck(_.pluck(cluster, 'membership'), 'checksum'))
+                if (!(checksums.length === 1 && typeof checksums[0] === 'number')) {
+                    assert.fail('not all checksums are equal')
+                }
+            }
+
+            cluster.forEach(function each(ringpop, i) {
+                ringpop.minProtocolPeriod = minProtocolPeriod;
             });
 
-            test(results, cluster, assert);
+            test(joinResults, cluster, assert);
+        }
+
+        var nodeExhausted = _.after(cluster.length, onConverged);
+        
+        cluster.forEach(function each(ringpop, i) {
+            ringpop.dissemination.on('changesExhausted', nodeExhausted);
+        });
+
+        cluster = bootstrapClusterOf(cluster, opts, function onBootstrap(results) {
+            if (typeof opts.tapAfterConvergence === 'function') {
+                opts.tapAfterConvergence(cluster);
```

Comment:
Nope, I have corrected it.

What do the mkNoGossip functions do exactly. The tests seem to pass wether they're on or commented out.